department-of-veterans-affairs/va.gov-cms
# Tugboat Next.js server can be restarted via Composer command

## Description

closes #17209 by adding commands to restart the next server after changing git branches.

## Testing done

Tested on Tugboat with a test branch that makes the next-build footer red and switched between that branch and others.

## Screenshots

`test-red-footer` branch on Tugboat

<img width="740" alt="Screenshot 2024-03-12 at 3 43 28 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3640707/8c72138e-f441-42b4-a744-5391d76b07b4">

## QA steps

1. Check that the preview server still works by going to the Tugboat URL for next, like https://next-suer27t71oqpgngyx21sblb9hukdweph.ci.cms.va.gov/.
   - [ ] Validate that the homepage loads and the footer is the typical "VA blue" or whatever they call it.
   - [ ] Validate any tricky parts of next-build, if you know them. I am not aware of typical smoke tests.
2. Log into the CMS and head to "/admin/content/deploy/next".
3. Fill out and submit the next server restart form with `test-red-footer` branch for next-build version and something else for the vets-website version.
4. Wait a few seconds and refresh. Refresh again if the form fields are not disabled and the build has not started.
   - [ ] Validate that the form fields are disabled during the build.
   - [ ] Validate that the `next-build.txt` file link leads to a log file with data in it. The log should grow every time you refresh and end with `==> Done`.
5. Refresh the tab on "/admin/content/deploy/next".
   - [ ] Validate that the form fields are no longer disabled.
6. Navigate back to the next preview server, a URL like https://next-suer27t71oqpgngyx21sblb9hukdweph.ci.cms.va.gov/.
   - [ ] Validate that the homepage loads and the footer is now bright red.
   - [ ] Validate any tricky parts of next-build, if you know them. I am not aware of typical smoke tests.
7. (Bonus Points) For a real parlor trick, you can navigate back to "/admin/content/deploy/next" and choose a different next-build branch to restart the server with. You'd have to create another PR in next-build with noticeable changes, but just reporting that users should be able to keep rebuilding the same Tugboat preview with various branches, even if most will only choose a branch one time.

Note: This PR, https://github.com/department-of-veterans-affairs/next-build/pull/451, should be deleted after testing. 

### Definition of Done


### Select Team for PR review
